prometheus_aws_guardduty_exporter/collector.py

```diff
@@ -9,6 +10,25 @@ class GuardDutyMetricsCollector():
     def __init__(self, regions: List[str]):
         self.regions = regions
         self.scrapeErrors = {region: 0 for region in regions}
+        self.botoConfig = None
+        self.lock = threading.Lock()
+
+    def scrape_metric_in_region(self, metric, region):
```

Comment:
Talking about (implicit) conventions, the method name should be camel case. Since it's "private", it should also be prefixed by `_`. Finally, being a "private" method, should be move to the end of the file.